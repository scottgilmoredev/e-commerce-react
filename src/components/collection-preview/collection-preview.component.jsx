import React from 'react';
import { withRouter } from 'react-router-dom';

// Components
import CollectionItem from '../collection-item/collection-item.component';

// Styles
import {
    CollectionPreviewContainer,
    PreviewContainer,
    TitleContainer,
  } from './collection-preview.styles';

/**
 * Display a preview of items (four) for the given collection.
 * @param {Array} items - list of collection items.
 * @param {String} title - collection title.
 */
export const CollectionPreview = ({ items, title, history, match, routeName }) => (
    <CollectionPreviewContainer>
        { /* Title */ }
        <TitleContainer onClick={() => history.push(`${ match.path }/${ routeName }`)}>
            {title.toUpperCase()}
        </TitleContainer>

        { /* Collection items */ }
        <PreviewContainer>
            {
                items
                    .filter((item, index) => index < 4)
                    .map((item) => (
                        <CollectionItem key={ item.id } item={ item } />
                    ))
            }
        </PreviewContainer>
    </CollectionPreviewContainer>
);

export default withRouter(CollectionPreview);
