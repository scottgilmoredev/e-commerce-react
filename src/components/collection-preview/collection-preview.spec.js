import React from 'react';
import { shallow } from 'enzyme';

// Component
import { CollectionPreview } from './collection-preview.component';

describe('CollectionPreview component', () => {
    let wrapper;
    let mockMatch;
    let mockHistory;
    const mockRouteName = 'hats';

    beforeEach(() => {
        mockMatch = {
            path: '/shop',
        };

        mockHistory = {
            push: jest.fn(),
        };

        const mockProps = {
            match: mockMatch,
            history: mockHistory,
            routeName: mockRouteName,
            title: 'hats',
            items: [],
        };

        wrapper = shallow(<CollectionPreview { ...mockProps } />);
    });

    // Render the component.
    it('should render the CollectionPreview component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    // Navigate to correct route.
    it('should call history.push with the right string when TitleContainer is clicked', () => {
        wrapper.find('TitleContainer').simulate('click');
        expect(mockHistory.push).toHaveBeenCalledWith(`${ mockMatch.path }/${ mockRouteName}`);
    });
});
