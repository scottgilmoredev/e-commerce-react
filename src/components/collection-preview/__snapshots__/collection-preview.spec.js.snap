// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should render the CollectionPreview component 1`] = `ShallowWrapper {}`;
